feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/transaction/filter/HistoryFiltersProviderFactory.kt

```diff
@@ -0,0 +1,14 @@
+package jp.co.soramitsu.feature_wallet_impl.presentation.transaction.filter
+
+class HistoryFiltersProviderFactory {
+    private var instance: HistoryFiltersProvider? = null
+
+    @Synchronized
+    fun get(): HistoryFiltersProvider {
+        if (instance != null) return instance!!
+
+        instance = HistoryFiltersProvider()
+
+        return instance!!
+    }
+}
```

Comment:
Why do we need this factory? We can use Dagger with `@ApplicationScope`. If you used Validator filters as a reference there it was required since their provider depends on some properties obtainable only in suspend manner, which is not what dagger can do.